src/detect-filesize.c

```diff
@@ -305,16 +289,17 @@ static int DetectFilesizeSetpTest01(void)
     DetectEngineCtx *de_ctx = NULL;
 
     res = DetectFilesizeInitTest(&de_ctx, &sig, &fsd, "1 <> 3 ");
-    if (res == 0) {
-        goto end;
-    }
+    FAIL_IF(res == 0);
 
     if(fsd == NULL)
         goto cleanup;
 
     if (fsd != NULL) {
-        if (fsd->arg1 == 1 && fsd->arg2 == 3 && fsd->mode == DETECT_UINT_RA)
-            res = 1;
+        FAIL_IF_NOT(fsd->arg1 == 1);
+        FAIL_IF_NOT(fsd->arg2 == 3);
+        FAIL_IF_NOT(fsd->mode == DETECT_UINT_RA);
+
+        PASS;
```

Comment:
Is it ever valid to have `fsd == NULL`? If not, then the test can be simplified further

Reply:
`fsd == NULL` leads to cleanup. Should I change it to `FAIL_IF_NULL` ?

Reply:
I'd say so.

Reply:
The cleanup steps are happening based on if `fsd==NULL` so should I just add them in sequence?

Reply:
Clean-up only has to be done in case of success, so you can put them in the very end, with the other clean-up steps.